### ISSUE TITLE
How should you do changeData on $interval for 0.5.1?

### ISSUE THREAD

**Issue by hilhorst** [l-lin/angular-datatables#528]
I set up my controller to get dtInstance through the two-way data binding as described in the v0.4.3 release notes.  When I add the following statement to the controller function (to do an automatic update of the table contents every 5 seconds) you see the "Loading..." message instead of the current contents of the table, and half of the time the vm.dtInstance object is logged as Object {}.

        $interval(function(){
                console.log(vm.dtInstance);
                if (typeof vm.dtInstance.changeData === 'function') vm.dtInstance.changeData( myPromise );
         }, 5000);

What would be the proper way to automatically update the contents of the table? Is this still possible in 0.5.1?

**Comment by l-lin** [comment]
Can you share your full code?

**Comment by hilhorst** [comment]
I'm stripping all things unrelated from my app so I can make a Plunk for it. This seems to be more work than I anticipated, but I hope to have one set up tomorrow.

**Comment by hilhorst** [comment]
I made a Plunk that shows the issue. On v0.5.0 the table is updated after the promise is resolved, on v0.5.1 the table is replaced with the "Loading..." until the promise is resolved. I built in an ugly delay to simulate the delay on the $resource call to emphasize the issue.

I put both in the same plunk. Only the version of angular-datatables is different, so other than the html file both are equal.

http://plnkr.co/edit/6uJZfn?p=preview
